src/settings/target/manifest.rs

```diff
@@ -52,27 +53,64 @@ impl Manifest {
 
     pub fn generate(
         name: String,
-        target_type: TargetType,
+        target_type: Option<TargetType>,
```

Comment:
why make this an option type when we end up falling back to a default in most cases?

Reply:
This is an `Option` because `--type` is an optional flag

Reply:
it's always been an optional flag, though; in the past we've simply fallen back to its default value before calling the command.

Reply:
Yes, this PR changes the point in the code where we determine the type. This is now just the value passed in by the flag, then we clone the repo, then we parse the toml and do the type check there. Any type passed in here will override a type that's in the template toml, and warn if it is different.